A4A: Update layout styling to match the Atelier design.

jkguidaven opened the pull request:
This PR updates the current A4A layout component styling to match the Atelier design.

https://github.com/Automattic/wp-calypso/assets/56598660/f77d2504-e854-4fb6-85bb-2a9fc22435b3



## Proposed Changes

* Update the Sidebar V2 component to use color theme properties to allow swappable colors using color scheme-defined colors.
* Update the A4A Layout component to have the correct background color, spacing, and borders that match the Atelier design.
* Update the placeholder page to use the new layout.

## Testing Instructions

* Switch branch: `git checkout update/a4a-layout-colors`
* Start the server by running `yarn start-a8c-for-agencies`.
* Go to http://agencies.localhost:3000/
* Confirm that the layout page matches the design.
* **Please also test if Jetpack cloud's sidebar is not broken with this PR.**

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

jkguidaven committed:
Update sidebar v2 styling with overridable colors.

jkguidaven committed:
Fix A4A sidebar colors.

jkguidaven committed:
Update a4a layout component to match Atelier design.

jkguidaven committed:
Fix issue with navigation tab alignment.

jkguidaven committed:
Fix spacing in body layout.

jkguidaven committed:
Update placeholders to use new layout.

jkguidaven commented on client/a8c-for-agencies/style.scss:
```diff
@@ -3,34 +3,42 @@
 	--a4a-corners-soft: 4px;
+
+	// FIXME: Remove this when we have a proper color scheme
+	--color-surface-backdrop: #021a23;
+
+	--color-sidebar-background: #021a23;
+	--color-sidebar-menu-hover-background: #02395c;
+	--color-sidebar-menu-hover-text: #fff;
+	--color-sidebar-menu-selected-background: #02395c;
+	--color-sidebar-menu-selected-text: #fff;
+	--color-sidebar-text: #fff;
+	--color-sidebar-text-alternative: #bbe0fa;
+	--color-sidebar-gridicon-fill: #bbe0fa;
```
We will remove these codes once we have a proper color scheme file for the A4A environment.

yashwin replied:
Are we also planning to move these colors to a file like this: https://github.com/Automattic/wp-calypso/blob/43eafeab20c2a990df379589a633ab02508d9b6e/packages/calypso-color-schemes/src/shared/color-schemes/_jetpack-cloud.scss

keoshi reviewed (approved):
This is looking fantastic, James! Thanks for pushing this forward so quickly too.

Only noticed two things. One that we can try to fix now, and another for more consideration with @jeffgolenski. Regardless, I think we can ship if we need to and adjust later.

#### Main panel height

The main panel is currently not taking the full height of the page. Not sure if a concern for right now or if it should be addressed in this PR, but worth noting nevertheless. This is what see at the bottom:

<img width="1452" alt="image" src="https://github.com/Automattic/wp-calypso/assets/390760/440ded7a-35d5-4837-b72e-8d8d18c81850">


Here's a prototype we did for Manage that has the right height and spacing from the sides: https://jetpack-manage-nav.vercel.app/

The related code can be found here, in case it's helpful: https://github.com/Automattic/jetpack-manage-nav-prototype/blob/trunk/src/lib/Content.svelte#L44-L55

#### Active/hover nav items

Right now there's no separation between `<li>` items in the sidebar, leading to a sequential active + hover (or vice-versa) to appear stuck together.

Proposing that we add a minimal 2px interval between these to create a clearer distinction. Here's a before and after of what it would look like.

Before | After
--|--
![image](https://github.com/Automattic/wp-calypso/assets/390760/bb40ab2d-7cff-40b1-a7a2-3e876d9f0d85) | ![image](https://github.com/Automattic/wp-calypso/assets/390760/4eb2f3ce-a749-4c24-9c35-9f34074a22a8)

---

Thanks again!

keoshi commented:
Sorry for not being super clear on the first item. Will add a screenshot to show what I see on my side.

jkguidaven committed:
Add 2px gap between sidebar menu items.

jkguidaven committed:
Address PR comments.

jkguidaven commented:
> Sorry for not being super clear on the first item. Will add a screenshot to show what I see on my side.

I updated the video to show both issues were fixed. 👍

keoshi commented:
Looks absolutely great, @jkguidaven ! Thanks again.

yashwin commented on client/a8c-for-agencies/components/a4a-logo/index.tsx:
```diff
@@ -1,7 +1,8 @@
 
-const LOGO_COLOR_PRIMARY = '#029CD7';
-const LOGO_COLOR_SECONDARY = '#021A23';
+export const LOGO_COLOR_PRIMARY = '#029CD7';
+export const LOGO_COLOR_SECONDARY = '#021A23';
+export const LOGO_COLOR_SECONDARY_ALT = '#fff';
```
Can we define these colors outside the logo component, maybe under `client/a8c-for-agencies`?

jkguidaven replied:
I don't think these colors are being used beyond the Logo component usage. For now, let's keep them here. If we end up using these colors elsewhere, we can move them to a global file.

yashwin commented on client/a8c-for-agencies/sections/sites/controller.tsx:
```diff
@@ -1,18 +1,45 @@
 	context.secondary = <SitesSidebar path={ context.path } />;
-	context.primary = <div>sites</div>;
+	context.primary = (
+		<Layout title="Overview" wide sidebarNavigation={ <MobileSidebarNavigation /> }>
```
Are we planning to update all the titles later? It says "Overview" for all the pages.

jkguidaven replied:
Good catch. this has been fixed 👍

yashwin reviewed (commented):
Good job on this, @jkguidaven! 

It looks good overall. I have left a few comments. Please let me know your thoughts.

jkguidaven committed:
Fix PR comments.

jkguidaven commented:
> Are we also planning to move these colors to a file like this: https://github.com/Automattic/wp-calypso/blob/43eafeab20c2a990df379589a633ab02508d9b6e/packages/calypso-color-schemes/src/shared/color-schemes/_jetpack-cloud.scss

Yes. Once the a8c brand is in Color studio, I plan to move it to a dedicated color scheme file.


Thanks for the review @yashwin I addressed your comments. Can you have another look? 🙏

yashwin commented on client/a8c-for-agencies/components/sidebar/header/index.tsx:
```diff
@@ -9,15 +9,21 @@ import { setLayoutFocus } from 'calypso/state/ui/layout-focus/actions';
+const AllSitesIcon = () => (
+	<A4ALogo
+		className="a4a-sidebar__all-sites-icon"
+		colors={ { primary: LOGO_COLOR_PRIMARY, secondary: LOGO_COLOR_SECONDARY_ALT } }
```
[Non-blocker] Sorry for not noticing it earlier. The default value for the `primaryColor` is `LOGO_COLOR_PRIMARY`; I think we can remove it.

yashwin reviewed (approved):
Thanks for addressing my comments.

Tested again & looks good to me.

jkguidaven committed:
Remove unncessary value.
